app/src/main/java/com/topjohnwu/magisk/core/su/SuRequestHandler.kt

```diff
@@ -52,42 +53,27 @@ class SuRequestHandler(
         return true
     }
 
-    private fun close() {
-        if (::output.isInitialized)
-            runCatching { output.close() }
-    }
-
     private suspend fun init(intent: Intent): Boolean {
         val uid = intent.getIntExtra("uid", -1)
-        if (uid <= 0) {
-            return false;
-        }
-        policy = SuPolicy(uid)
         val pid = intent.getIntExtra("pid", -1)
-        if (pid <= 0) {
-            return false;
+        val fifo = intent.getStringExtra("fifo")
+        if (uid <= 0 || pid <= 0 || fifo == null) {
+            return false
         }
-        val fifo = intent.getStringExtra("fifo") ?: "/dev/socket/magisk_su_request_$pid"
-
         try {
-            output = DataOutputStream(FileOutputStream(fifo))
-            try {
-                pkgInfo = pm.getPackageInfo(uid, pid) ?: PackageInfo().apply {
-                    val name = pm.getNameForUid(uid) ?: throw PackageManager.NameNotFoundException()
-                    // We only fill in sharedUserId and leave other fields uninitialized
-                    sharedUserId = name.split(":")[0]
-                }
-            } catch (e: PackageManager.NameNotFoundException) {
-                Timber.e(e)
-                respond(SuPolicy.DENY, -1)
-                return false
+            pkgInfo = pm.getPackageInfo(uid, pid) ?: PackageInfo().apply {
+                val name = pm.getNameForUid(uid) ?: throw PackageManager.NameNotFoundException()
+                // We only fill in sharedUserId and leave other fields uninitialized
+                sharedUserId = name.split(":")[0]
             }
-            return true
-        } catch (e: IOException) {
+        } catch (e: PackageManager.NameNotFoundException) {
             Timber.e(e)
-            close()
+            respond(SuPolicy.DENY, -1)
```

Comment:
This `respond` will crash because `this.output` was not set.

Reply:
fixed